### PR TITLE
NO-JIRA make automountServiceAccountToken in _pod.tpl configurable

### PR DESCRIPTION
**Pull request by vladimirddimov**
Please ensure your pull request adheres to the following guidelines:
- [x] explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [x] Document your Changes in the `CHANGELOG.md` file of the respected chart as well as the `Chart.yaml`

Our setup of SonarQube is using CloudSQL and the authentication is going via CloudSQL Proxy which needs automountServiceAccountToken to be set to true in order ot be able to authenticate via IAM Service Account.

### CODE REVIEW

**Review by marcvn-eng — APPROVED** [review]

**Comment by jCOTINEAU** [comment]
Hello @vladimirddimov, could i ask you to rebase and propagate the fix also to the DCE chart?

Also the changes are to be set in 10.7.0 as we are releasing at the same time as the upcoming SonarQube version.

Thanks a lot again for your contribution

File: charts/sonarqube/Chart.yaml [hunk 1]
```diff
@@ -32,6 +32,8 @@ annotations:
     - name: Chart Source
       url: https://github.com/SonarSource/helm-chart-sonarqube/tree/master/charts/sonarqube
   artifacthub.io/changes: |
+    - kind: changed
```
Comment (davividal) [inline charts/sonarqube/Chart.yaml:35]: Please put your changes at the bottom.

File: charts/sonarqube/templates/_pod.tpl [hunk 2]
```diff
@@ -19,7 +19,7 @@ metadata:
     {{- toYaml . | nindent 4 }}
     {{- end }}
 spec:
-  automountServiceAccountToken: false
+  automountServiceAccountToken: {{ .Values.podSpec.automountServiceAccountToken }}
```
Comment (davividal) [inline charts/sonarqube/templates/_pod.tpl:22]: Thank you for your contribution!

Can you please replicate your changes to the `sonarqube-dce` chart?

**Review by davividal — CHANGES_REQUESTED** [review]

**Comment by jCOTINEAU** [comment]
Hey @davividal, considering this PR has not received any updates, could you either close it or take over and build on top of the contribution?

**Comment by carminevassallo** [comment]
As we don't see further reaction on this PR, we decided to create [a hardening ticket](https://sonarsource.atlassian.net/browse/SONAR-23395) about this feature.

Thanks again @vladimirddimov for your contribution.

**Pull request closed without merging**